File: src/main/java/com/kingmang/ixion/ast/ParameterStatement.java
package com.kingmang.ixion.ast;

import com.kingmang.ixion.StatementVisitor;
import com.kingmang.ixion.lexer.Position;
import com.kingmang.ixion.lexer.Token;

public final class ParameterStatement extends Statement {
    public final Token name;
    public final TypeStatement type;

    public ParameterStatement(Position pos, Token name, TypeStatement type) {
        super(pos);
        this.name = name;
        this.type = type;
    }

    public <R> R accept(StatementVisitor<R> visitor) {
        return visitor.visitParameterStmt(this);
    }
}

File: src/main/java/com/kingmang/ixion/ast/ParameterStatement.kt
package com.kingmang.ixion.ast

import com.kingmang.ixion.StatementVisitor
import com.kingmang.ixion.lexer.Position
import com.kingmang.ixion.lexer.Token

class ParameterStatement(pos: Position?, @JvmField val name: Token?, @JvmField val type: TypeStatement?) : Statement(pos) {

    override fun <R> accept(visitor: StatementVisitor<R?>?): R? {
        return visitor?.visitParameterStmt(this)
    }
}